### PR TITLE
EASY-2273: loading, saving and submitting rejected deposits

### PR DESCRIPTION
**Pull request by rvanheest**
Fixes EASY-2273

#### When applied it will
* fetch the deposit's state before loading its metadata and files
* implement turning a 'rejected deposit' back into a 'draft deposit'
* create a fallback page for when the deposit could not be found (404 response)
* create a fallback page for when the server could not be reached (500/503 response)
* create fallback pages for when the deposit is already submitted/in review/archived
* adjust and improve mock data in the local server
* (outside the issue) improve the error message shown when the deposit overview cannot be displayed (due to 500/503 response)
* (outside the issue) improve the title of the deposit overview page

#### During the review process we need to consider
* [ ] styling of the new fallback pages; currently nothing more than the default styling that was already present applies, but maybe we should/can improve on that?

**_response 200 - SUBMITTED_**
...

**_response 200 - IN REVIEW_**
...

**_response 200 - ARCHIVED_**
...

**_response 404 - deposit not found/not your deposit_**
...

**_response 500/503 - `easy-deposit-api` turned off_**
image

@DANS-KNAW/easy for review

### CODE REVIEW

**Comment by jo-pol** [comment]
The webui has light blue for feedback messages that are not errors, as on:
https://drivenbydata.atlassian.net/57a048b2-96c6-584e-86df-fb977af20a3b
For what it's worth, I'd say: apply that style to the line "has been submitted / is in review / has been published". Thus the page experience get closer to a submit form, yet is different enough. 

I'd expect a pink line on the 404 page like on the 500/503 (or whenever there emerges a difference in presentation between user errors and system errors: like missing mandatory fields when you try to submit a deposit)

File: src/main/typescript/components/overview/DepositOverviewPage.tsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ import "../../../resources/css/depositOverviewPage"
 
 const DepositOverviewPage = () => (
     <>
-        <h1>Deposit Overview Page</h1>
+        <h1>My Deposits</h1>
```
Comment (jo-pol) [inline src/main/typescript/components/overview/DepositOverviewPage.tsx]: 👍

File: src/test/typescript/mockserver/deposit.ts [hunk 2]
```diff
@@ -50,7 +50,7 @@ export const depositData3: Deposit = {
 export const depositData4: Deposit = {
     title: "Old Dataset",
     state: "ARCHIVED",
-    stateDescription: "easy-dataset:id",
+    stateDescription: "The dataset is published at <a href='https://deasy.dans.knaw.nl/ui/datasets/id/easy-dataset:11' target='_blank'>https://deasy.dans.knaw.nl/ui/datasets/id/easy-dataset:11</a>",
```
Comment (jo-pol) [inline src/test/typescript/mockserver/deposit.ts]: If this is also used in the deposit-overview page, the deposit-api would no longer need to construct the link and the application properties of the deposit-api can forget its knowledge about the webui.

Comment (rvanheest) [reply src/test/typescript/mockserver/deposit.ts]: this is just a test object... The API should serve a similar kind of object.

**Pull request merged**